src/compiler/parser.ts

```diff
@@ -1021,6 +1052,121 @@ namespace ts {
             return node;
         }
 
+        function reparseTopLevelAwait(sourceFile: SourceFile) {
+            return visitEachChild(sourceFile, visitor, reparseContext);
+
+            function visitor(node: Node): VisitResult<Node> {
+                if (!(node.transformFlags & TransformFlags.ContainsPossibleTopLevelAwait)) {
+                    return node;
+                }
+                switch (node.kind) {

```

Comment:
This is essentially every non-expression node which contains an expression node, yeah? Can we add a comment to that effect? The requirement for inclusion here is nonobvious.